controllers/create_metrics_monitor.go

```diff
@@ -0,0 +1,159 @@
+			APIVersion: rbacv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "search-metrics-monitor",
+			Namespace: instance.GetNamespace(),
+		},
+		Rules: getRules(),
+	}
+	err := controllerutil.SetControllerReference(instance, cr, r.Scheme)
+	if err != nil {
+		log.Info("Could not set control for Role ", "name", "search-metrics-monitor")
+	}
+	return cr
+}
+
+func (r *SearchReconciler) MetricsRoleBinding(instance *searchv1alpha1.Search) *rbacv1.RoleBinding {
+	crb := &rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "RoleBinding",
+			APIVersion: rbacv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "search-metrics-monitor",
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     "search-metrics-monitor",
+			APIGroup: rbacv1.GroupName,
+		},
+		Subjects: []rbacv1.Subject{{
+			Kind:      "ServiceAccount",
+			Name:      "prometheus-k8s",
+			Namespace: "openshift-monitoring",
+		},
+		},
+	}
+	err := controllerutil.SetControllerReference(instance, crb, r.Scheme)
+	if err != nil {
+		log.Info("Could not set control for RoleBinding", "name", "search-metrics-monitor")
+	}
+	return crb
+}
+
+func (r *SearchReconciler) ServiceMonitor(instance *searchv1alpha1.Search, deployment string) *monitorv1.ServiceMonitor {
+	smName := deployment + "-monitor"
+	cr := &monitorv1.ServiceMonitor{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ServiceMonitor",
+			APIVersion: monitorv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      smName,
+			Namespace: "openshift-monitoring",
+		},
+		Spec: monitorv1.ServiceMonitorSpec{
+			JobLabel:          deployment,
+			NamespaceSelector: monitorv1.NamespaceSelector{MatchNames: []string{instance.GetNamespace()}},
+			Endpoints: []monitorv1.Endpoint{
+				{
+					Port:            deployment,
+					Scheme:          "https",
+					ScrapeTimeout:   "10s",
+					Interval:        "60s",
+					BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+				},
+			},
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"search-monitor": deployment}},
+		},
+	}
+	err := controllerutil.SetControllerReference(instance, cr, r.Scheme)
+	if err != nil {
+		log.Info("Could not set control for ServiceMonitor ", "name", smName)
+	}
+	return cr
+}
+
+func (r *SearchReconciler) createServiceMonitor(ctx context.Context,
+	smonitor *monitorv1.ServiceMonitor,
+) (*reconcile.Result, error) {
+
+	found := &monitorv1.ServiceMonitor{}
+	err := r.Get(ctx, types.NamespacedName{
+		Name:      smonitor.Name,
+		Namespace: smonitor.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+		err = r.Create(ctx, smonitor)
+		if err != nil {
+			log.Error(err, "Could not create servicemonitor "+smonitor.Name)
+			return &reconcile.Result{}, err
+		}
+		log.Info("Created servicemonitor" + smonitor.Name)
+		log.V(9).Info("Created  servicemonitor ", "name", smonitor.Name)
```

Comment:
These 2 logs print the same info.  Did we intend to print the entire `smonitor` object instead?
I noticed a similar pattern at the top of this file.

Reply:
Yes, thanks.